More examples for using slice() (#35226)
mdn/content: files/en-us/web/javascript/reference/global_objects/array/slice/index.md

// fruits contains ['Banana', 'Orange', 'Lemon', 'Apple', 'Mango']
// citrus contains ['Orange','Lemon']
```

> In this example, `slice(1, 3)` extracts elements from index `1` up to, but not including, index `3`, resulting in a new array `['Orange', 'Lemon']`.

Author: Resolved, add one line space.

> In this example, `slice(1, 3)` extracts elements from index `1` up to, but not including, index `3`, resulting in a new array `['Orange', 'Lemon']`.

Author: Resolved, adding one line space.

### Example omitting the `end` parameter

```js

> const fruits = ['Apple', 'Banana', 'Orange', 'Mango', 'Pineapple'];

Author: Resoled, adding double quotes.

const tropical = fruits.slice(2);
console.log(tropical); // ['Orange', 'Mango', 'Pineapple']
```

In this example, `slice(2)` extracts elements from index `2` to the end of the array.

### Example with negative indices

```js

> const fruits = ['Apple', 'Banana', 'Orange', 'Mango', 'Pineapple'];

Author: Resoled, adding double quotes.

const lastTwo = fruits.slice(-2);
console.log(lastTwo); // ['Mango', 'Pineapple']
```

> In this example, if negative indices are used, slice counts from the end of the array. In this case, `slice(-2)` extracts the last two elements of the array.

Author: Resolved, adding one line space.

When using a negative index with the slice method, negative indices are counted from the end of the array, starting at `-1` for the last element, `-2` for the second-to-last element, and so on.

When using slice with a **single negative index**, the negative index is included because it is the starting point of the extraction.

```
|     |     |     |     |     |
|  S  |  L  |  I  |  C  |  E  |
|     |     |     |     |     | 
  -5    -4    -3    -2    -1   

<--- read from reverse
```

> ### Example using slice with a positive start index and a negative end index 

Author: Quit space

```js
const fruits = ['Apple', 'Banana', 'Orange', 'Mango', 'Pineapple'];

// Using positive start index and negative end index
const sliceExample = fruits.slice(1, -1);
console.log(sliceExample); // ['Banana', 'Orange', 'Mango']
```

> In this example, `slice(1, -1)` starts extracting from index `1` and goes up to, but does not include, the element at index `-1` (which is the last element). This results in a new array with `['Banana', 'Orange', 'Mango']`.

Author: Adding one line space.

When using slice with a **positive index and a negative index**, the extraction starts at the positive index and continues up to, but does not include, the negative index. The negative index is not included because the slice method always excludes the element at the final index specified, regardless of whether it is positive or negative.

```
read from start --->

   0     1     2     3     4
|     |     |     |     |     |
|  S  |  L  |  I  |  C  |  E  |
|     |     |     |     |     | 

>   -5    -4    -3    -2    -1   

Author: Quit blank space.

|  S  |  L  |  I  |  C  |  E  |
|     |     |     |     |     |
  -5    -4    -3    -2    -1

<--- read from reverse
```

### Using a positive start index and a negative end index

```js
const fruits = ["Apple", "Banana", "Orange", "Mango", "Pineapple"];

// Using positive start index and negative end index
const sliceExample = fruits.slice(1, -1);
console.log(sliceExample); // ['Banana', 'Orange', 'Mango']
```

In this example, `slice(1, -1)` starts extracting from index `1` and goes up to, but does not include, the element at index `-1` (which is the last element). This results in a new array with `['Banana', 'Orange', 'Mango']`.

When using `slice` with a positive index and a negative index, the extraction starts at the position of the positive index and continues up to, but does not include, the position of the negative index. The `slice` method always excludes the element at the final index specified, regardless of whether it is positive or negative.

```plain
read from start --->

   0     1     2     3     4
|     |     |     |     |     |
|  S  |  L  |  I  |  C  |  E  |
|     |     |     |     |     |
  -5    -4    -3    -2    -1

<--- read from reverse
```

> ### Using slice

Reviewer: ```suggestion
### Using slice with arrays of objects
```

// citrus contains ['Orange','Lemon']
```

In this example, `slice(1, 3)` extracts elements from index `1` up to, but not including, index `3`, resulting in a new array `['Orange', 'Lemon']`.

> ### Example omitting the `end` parameter

Reviewer: ```suggestion
### Omitting the end parameter
```

```js
const fruits = ["Apple", "Banana", "Orange", "Mango", "Pineapple"];

const tropical = fruits.slice(2);
console.log(tropical); // ['Orange', 'Mango', 'Pineapple']
```

In this example, `slice(2)` extracts elements from index `2` to the end of the array.

> ### Example with negative indices

Reviewer: ```suggestion
### Using negative indices
```

```js
const fruits = ["Apple", "Banana", "Orange", "Mango", "Pineapple"];

const lastTwo = fruits.slice(-2);
console.log(lastTwo); // ['Mango', 'Pineapple']
```

> In this example, if negative indices are used, slice counts from the end of the array. In this case, `slice(-2)` extracts the last two elements of the array.
> 
> When using a negative index with the slice method, negative indices are counted from the end of the array, starting at `-1` for the last element, `-2` for the second-to-last element, and so on.
> 
> When using slice with a **single negative index**, the negative index is included because it is the starting point of the extraction.

Reviewer: ```suggestion
In this example, `slice(-2)` extracts the last two elements of the array. When using a negative index with the `slice` method, negative indices are counted from the end of the array, starting at `-1` for the last element, `-2` for the second-to-last element, and so on. The negative index `-2` itself is included because it is the starting point of the extraction.
```

```plain
|     |     |     |     |     |
|  S  |  L  |  I  |  C  |  E  |
|     |     |     |     |     |
  -5    -4    -3    -2    -1

<--- read from reverse
```

> ### Example using slice with a positive start index and a negative end index

Reviewer: ```suggestion
### Using a positive start index and a negative end index
```

```js
const fruits = ["Apple", "Banana", "Orange", "Mango", "Pineapple"];

// Using positive start index and negative end index
const sliceExample = fruits.slice(1, -1);
console.log(sliceExample); // ['Banana', 'Orange', 'Mango']
```

In this example, `slice(1, -1)` starts extracting from index `1` and goes up to, but does not include, the element at index `-1` (which is the last element). This results in a new array with `['Banana', 'Orange', 'Mango']`.

> When using slice with a **positive index and a negative index**, the extraction starts at the positive index and continues up to, but does not include, the negative index. The negative index is not included because the slice method always excludes the element at the final index specified, regardless of whether it is positive or negative.

Reviewer: ```suggestion
When using `slice` with a positive index and a negative index, the extraction starts at the position of the positive index and continues up to, but does not include, the position of the negative index. The `slice` method always excludes the element at the final index specified, regardless of whether it is positive or negative.
```

const tropical = fruits.slice(2);
console.log(tropical); // ['Orange', 'Mango', 'Pineapple']
```

In this example, `slice(2)` extracts elements from index `2` to the end of the array.

### Using negative indices

```js
const fruits = ["Apple", "Banana", "Orange", "Mango", "Pineapple"];

const lastTwo = fruits.slice(-2);
console.log(lastTwo); // ['Mango', 'Pineapple']
```

In this example, `slice(-2)` extracts the last two elements of the array. When using a negative index with the `slice` method, negative indices are counted from the end of the array, starting at `-1` for the last element, `-2` for the second-to-last element, and so on. The negative index `-2` itself is included because it is the starting point of the extraction.

```plain
|     |     |     |     |     |
|  S  |  L  |  I  |  C  |  E  |
|     |     |     |     |     |
  -5    -4    -3    -2    -1

<--- read from reverse
```

### Using a positive start index and a negative end index

```js
const fruits = ["Apple", "Banana", "Orange", "Mango", "Pineapple"];

// Using positive start index and negative end index
const sliceExample = fruits.slice(1, -1);
console.log(sliceExample); // ['Banana', 'Orange', 'Mango']
```

> In this example, `slice(1, -1)` starts extracting from index `1` and goes up to, but does not include, the element at index `-1` (which is the last element). This results in a new array with `['Banana', 'Orange', 'Mango']`.
> 
> When using `slice` with a positive index and a negative index, the extraction starts at the position of the positive index and continues up to, but does not include, the position of the negative index. The `slice` method always excludes the element at the final index specified, regardless of whether it is positive or negative.

Reviewer: ```suggestion
In this example, `slice(1, -1)` starts extracting from index `1` and goes up to, but does not include, the element at index `-1` (which is the last element). This results in a new array with `['Banana', 'Orange', 'Mango']`. The `slice` method always excludes the element at the final index specified, regardless of whether it is positive or negative.
```
